core/src/main/java/org/apache/calcite/prepare/RelOptTableImpl.java

```diff
@@ -118,9 +115,9 @@ public static RelOptTableImpl create(
       RelDataType rowType,
       List<String> names,
       Table table,
-      Expression expression) {
+      Function<Class, Expression> expressionFunction) {
```

Comment:
This is a breaking change to a public method. (Albeit a method that has no javadoc.) We need to keep the old method (deprecated) for compatibility.

New method needs javadoc.

Reply:
Added the method with the previous signature, marked it as deprecated and added JavaDoc for a new one.